tests/integration/test_storage_s3_queue/test.py

```diff
@@ -847,107 +848,85 @@ def test_max_set_age(started_cluster):
     )
     create_mv(node, table_name, dst_table_name)
 
-    total_values = generate_random_files(
+    _ = generate_random_files(
         started_cluster, files_path, files_to_generate, row_num=1
     )
 
-    expected_rows = 10
+    expected_rows = files_to_generate
 
     node.wait_for_log_line("Checking node limits")
     node.wait_for_log_line("Node limits check finished")
 
     def get_count():
         return int(node.query(f"SELECT count() FROM {dst_table_name}"))
 
-    for _ in range(20):
-        if expected_rows == get_count():
-            break
-        time.sleep(1)
+    def wait_for_condition(check_function, max_wait_time=30):
+        before = time.time()
+        while time.time() - before < max_wait_time:
+            if check_function():
+                return
+            time.sleep(0.1)
+        assert False
 
-    assert expected_rows == get_count()
-    assert 10 == int(node.query(f"SELECT uniq(_path) from {dst_table_name}"))
+    wait_for_condition(lambda: get_count() == expected_rows)
+    assert files_to_generate == int(node.query(f"SELECT uniq(_path) from {dst_table_name}"))
 
     time.sleep(max_age + 5)
 
-    expected_rows = 20
-
-    for _ in range(20):
-        if expected_rows == get_count():
-            break
-        time.sleep(1)
-
-    assert expected_rows == get_count()
-    assert 10 == int(node.query(f"SELECT uniq(_path) from {dst_table_name}"))
+    expected_rows *= 2
+    wait_for_condition(lambda: get_count() == expected_rows)
+    assert files_to_generate == int(node.query(f"SELECT uniq(_path) from {dst_table_name}"))
 
     paths_count = [
         int(x)
         for x in node.query(
             f"SELECT count() from {dst_table_name} GROUP BY _path"
         ).splitlines()
     ]
-    assert 10 == len(paths_count)
+    assert files_to_generate == len(paths_count)
     for path_count in paths_count:
         assert 2 == path_count
 
-    failed_count = int(
-        node.query(
+    def get_object_storage_failures():
+        return int(node.query(
             "SELECT value FROM system.events WHERE name = 'ObjectStorageQueueFailedFiles' SETTINGS system_events_show_zero_values=1"
-        )
-    )
+        ))
+
+    failed_count = get_object_storage_failures()
 
     values = [
         ["failed", 1, 1],
     ]
     values_csv = (
         "\n".join((",".join(map(str, row)) for row in values)) + "\n"
     ).encode()
-    put_s3_file_content(started_cluster, f"{files_path}/fff.csv", values_csv)
+    file_with_error = f"fff_{uuid4().hex}.csv"
```

Comment:
Don't we use unique containers for every test and a unique Minio instance for every container?

Reply:
If not, you can also fix this test: https://github.com/ClickHouse/ClickHouse/pull/66802/files

Reply:
That's right AFAIU, but I couldn't use `--count 1000` to run successfully a bunch of tests sequentially without this modification. As expected, between each test execution ClickHouse's system tables are kept, so the `count()` check against `system.s3queue_log` fails if we always use the same filename.

I'll try to investigate how I can run in parallel N docker containers running the same test, because I tried but failed because some files are apparently shared by pytest.